tests/integration/test_fim/test_files/test_wildcards_complex/test_wildcards_complex_runtime.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (C) 2015-2021, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import os
+import sys
+import pytest
+from wazuh_testing import global_parameters
+from wazuh_testing import fim
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.configuration import load_wazuh_configurations, check_apply_test
+from wazuh_testing.tools.monitoring import FileMonitor
+
+# Marks
+
+pytestmark = pytest.mark.tier(level=1)
+
+# Variables
+frequency_scan = 10
+test_folder = os.path.join(PREFIX, 'test_folder')
+test_directories = [test_folder]
+
+matched_dirs = [os.path.join('stardir', 'sub_test'), os.path.join('multiple_wildcards', 'sub_test'),
+                os.path.join('directory_test', 'test_subdir1'), os.path.join('test_all', 'testdir'),
+                os.path.join('test_all', 'testdir', 'all')]
+
+no_match_dirs = ['random_directory']
+
+wildcards = ','.join([os.path.join(test_folder, 'star*', 'sub*'), os.path.join(test_folder, 'mul*', '*test'),
+                      os.path.join(test_folder, '*test*', '*dir?'), os.path.join(test_folder, 'test_all', '*'),
+                      os.path.join(test_folder, 'test_all', '*', '*')])
+
+wazuh_log_monitor = FileMonitor(fim.LOG_FILE_PATH)
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf_wildcards_runtime.yml')
+test_folders = matched_dirs + no_match_dirs
+
+# Configurations
+
+conf_params = {'TEST_WILDCARDS': wildcards, 'FREQUENCY': frequency_scan}
+parameters, metadata = fim.generate_params(extra_params=conf_params)
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+
+
+# Fixtures
+
+@pytest.fixture()
+def wait_for_initial_scan():
+    """Fixture that waits for the initial scan, independently of the configured mode."""
+    fim.detect_initial_scan(wazuh_log_monitor)
+
+
+@pytest.fixture()
+def create_test_folders():
+    """Fixture that creates all the folders specified in the `test_subdirectories` list"""
+    for dir in test_folders:
+        split_path = os.path.split(dir)
+        parent_folder = os.path.join(test_folder, split_path[0])
+        if not os.path.exists(parent_folder):
+            os.mkdir(os.path.join(parent_folder))
+        if not os.path.exists(os.path.join(parent_folder, split_path[1])):
+            os.mkdir(os.path.join(parent_folder, split_path[1]))
```

Comment:
This should delete those directories at the end.

Reply:
These directories are removed after the test has finished, as all the subfolders are created inside the folder specified in the `test_directories` list